README.md

```diff
@@ -1,3 +1,156 @@
+With filtering, the developers can provide the unified spec, along with a list of filter criteria that
+they want to match, and get a new spec matching the criteria. For example, filtering would allow the developers
+to generate a new spec with only the "public" tagged operations.
+
+#### Overlay
+Overlay allows developers to place OpenAPI spec fragments on top of existing specs.
+
+For example, in the case of both filtering and merging, the resultant spec represents a different API product
+than the input specs. This is why overlaying functionality will also be supported. An overlay can
+be applied using a “default file” which contains metadata about the final API product that is
+produced from Spec Math operations. For example, if a company wishes to build a web
+application using filtered pieces of their unified API, the original spec will no longer have an
+accurate `info: title` key path. The overlay will contain something like `info: title: Company Web
+Application` which will be applied to the resultant specification.
+
+### Overview
+
+This section provides descriptions of Spec Math operations. See the background section for an example of how
+some of these capabilities of Spec Math might be useful.   
+
+For additional specific examples of each operation described below, please take a look at the integration
+tests for the [Spec Math Library](library/src/test/java/org/specmath/library/SpecMathTest.java). Tests are
+provided for some example usages of the operations discussed below. Each test contains a sample input and output.
+
+#### Operations
+
+##### Union
+
+##### Union Of Two Specs
+Union of two specs takes specs A and B along with an optional set of rules to create a new spec C which contains A and B.
+
+A collision occurs when two YAML key paths have different primitive values. For example, if
+spec A had `info: license: name: MIT` and spec B had `info: license: name: GPL`, there would be
+a conflict in the `info: license: name` key path. To resolve this, the user can provide a defaults
+file as an overlay. In cases where the overlay cannot resolve the conflict, the conflicting key
+paths will be reported back. The user can then resolve the conflicts by either:
+
+- using a conflict resolution file ([sample](library/src/test/resources/conflictMerged.yaml)) passed in as a parameter to the union,
+- manually resolving the specs by removing the conflict
+- or updating the defaults file.
+
+##### Union Of Multiple Specs
+The requirements for union of multiple specs are still being determined. 
+
+#### Overlay
+Overlay takes spec A along with a defaults file D ([sample](library/src/test/resources/metadata.yaml)) to modify spec A. D is a spec fragment
+which will be placed "on top" of spec A. 
+In the case of a collision, the defaults file takes priority. 
+An overlay operation is a specialized Union operation between A and D where D will always take priority.
+Therefore, Overlay operations will not have conflicts. 
```

Comment:
What if the overlay refers to spec elements that don't exist in the main spec? That might not be a conflict (by definition) but it could be an error.

Reply:
good point, I will clarify that